### PR TITLE
Sports Scores Refactoring & NFL

### PR DESCRIPTION
**Pull request by sdougbrown**
Expanding the MLB instant answer into multiple sports requires some changes in order to reduce the amount of duplicated code.  This generalizes some of the basics, and implements an NFL boxscore using sample data.

This requires a change to our internal architecture so this is not ready to be merged yet, by any means. :smile:

### CODE REVIEW

File: share/spice/sports/nfl/games/sports_nfl_games.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        DDG.require('sports', function(){
+            Games = env.ddg_spice_sports_games;
+            
+            Games.init(NFLInit);
+        });
+        
+        var NFLInit = function () {
+                Spice.add({
+                    id: 'nfl_games',
+                    name: 'NFL Games',
```
Comment (b1ake) [inline share/spice/sports/nfl/games/sports_nfl_games.js]: @sdougbrown What do you think about renaming the tabs to just `NFL`? @nilnilnil suggested.

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 2]
```diff
@@ -7,147 +9,88 @@
...
+        DDG.require('sports', function(){
+            Games = env.ddg_spice_sports_games;
+                
+            Games.init(MLBInit);
+        });
+        
+        var MLBInit = function () {
+                Spice.add({
+                    id: 'mlb_games',
+                    name: 'MLB Games',
```
Comment (b1ake) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: @sdougbrown Same here, maybe rename to `MLB`?

Comment (b1ake) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: cc @nilnilnil

**Comment by bsstoner** [comment]
:+1:

**Comment by sdougbrown** [comment]
:boom:

**Pull request merged**